packages/core/src/service/services/collection.service.ts

```diff
@@ -364,6 +364,8 @@ export class CollectionService implements OnModuleInit {
                 collectionIds: [collection.id],
             });
         }
```

Comment:
If `input.filters` is populated, then a `CollectionModificationEvent` will be published twice - once here on line 368, and also once on line 105 after the "apply-collection-filters" job has complemented.

Therefore I think lines 367 & 368 should be inside an `else {} ` block.

Reply:
It was my first implementation :) But this doesn't work. Cause ApplyFilterInternal method checks only new/deleted products.

Without else block everything works as expected, because search plugins reduces ids here: https://github.com/vendure-ecommerce/vendure/blob/6f7cc34acda54693fc3ab33cf9f209ecabfc45cb/packages/elasticsearch-plugin/src/plugin.ts#L325

Reply:
Ah ok that makes sense. In that case, would it make sense to at least add a call to `unique()` in the plugin, since in this case there will likely be duplicate variant IDs sent with the 2 events? (if so, would also apply to the DefaultSearchPlugin)

I've also got a nagging feeling that there is a potentially problematic race condition here:

1. The "apply-collection-filters" job is enqueued.
2. The CollectionModificationEvent is published with the _current_ variants associated with the Collection
3. The job gets processed, updates the variants based on the filter changes, and then publishes another CollectionModificationEvent, with the new set of variant ids.

I _think_ it's actually ok, because once all these variant IDs arrive at the search indexer, it will grab the fresh, up-to-date data on each variant. Just thinking "out loud" here - maybe you can spot any issues with this flow?

Reply:
I changed logic of `applyCollectionFiltersInternal` and `update`.

If `input.filters` is provided in `update` method, we settle job with `applyToChangedVariantsOnly=false`.
`applyCollectionFiltersInternal` checks `applyToChangedVariantsOnly` variable and if it equals `true` (default) - then only new/deleted variants are processed. if it equals `false` - then all variants are processed.

If `input.filters` is not provided in `update` method hen all variants are processed.